http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -782,31 +787,60 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
       for {
         ctx <- ev.ask
         wsmApi <- buildWsmControlledResourceApiClient
-        controlledDbsForApp = appInstall.databases.collect { case d @ ControlledDatabase(_, _, _) => d }
-        // retrieve databases that might already be created in workspace
+
+        // get a list of database types required for this app
+        controlledDbsRequiredForApp = appInstall.databases.collect { case d @ ControlledDatabase(_, _, _) => d }
+
+        // retrieve set of databases WSM has created for this workspace
         existingControlledDbsInWorkspace <- retrieveWsmDatabases(wsmResourceApi,
-                                                                 controlledDbsForApp.map(_.prefix).toSet,
+                                                                 controlledDbsRequiredForApp.map(_.prefix).toSet,
                                                                  workspaceId.value
         )
-        wsmControlledDBResources <- controlledDbsForApp
+
+        // verify or create a list of APP_CONTROLLED_RESOURCE records for each required database, creating the db if needed
+        wsmControlledDBResources <- controlledDbsRequiredForApp
           .map { controlledDbForApp =>
-            // if a database already exists (because of workspace cloning) use that otherwise create a new one
-            if (
-              existingControlledDbsInWorkspace
-                .exists(existingDb => controlledDbForApp.prefix == existingDb.wsmDatabaseName)
-            ) {
-              F.pure(
-                existingControlledDbsInWorkspace
-                  .find(clonedDatabase => controlledDbForApp.prefix == clonedDatabase.wsmDatabaseName)
-                  .get
-              )
-            } else {
-              createWsmDatabaseResource(app, workspaceId, controlledDbForApp, namespacePrefix, owner, wsmApi).map {
-                db =>
-                  WsmControlledDatabaseResource(db.getAzureDatabase.getMetadata.getName,
-                                                db.getAzureDatabase.getAttributes.getDatabaseName
+            val maybeExistingDb = existingControlledDbsInWorkspace
+              .find(existingDb => controlledDbForApp.prefix == existingDb.wsmDatabaseName)
+
+            maybeExistingDb match {
+              // a database already exists (because of workspace cloning) verify or create a APP_CONTROLLED_RESOURCE for it
+              case Some(existingDb) =>
+                // there is an existing db, so check to see if we have previously mapped it to appControlledResource
+                val resourceId = existingDb.controlledResourceId.get
```

Comment:
This would crash if `controlledResourceId` is `None` -- what made me think this shouldn't be an Option :)